### PR TITLE
W820NB supported

### PR DESCRIPTION
**Pull request by BreakpointDebug1234**
Tested W820NB as a generic device. Everything seems to work except the sound effect panel which doesn't seem to do anything.

Tested with firmware version: 01.00.01

I had to use google translate for Simplified Chinese readme.

### CODE REVIEW

**Comment by wh201906** [comment]
Thanks for your work! I can add a dedicated profile for W820NB before merging this PR.
According to some reviews, it seems W820NB doesn't support Sound Effect so it's reasonable these functions don't work.

I also found the `Control Settings` panel might be unavailable for W820NB. Could you please help me to test it?
(For W820NB Double Gold, when clicking the Bluetooth button, the modes will only switch among the selected modes in the `Control Settings`)

**Comment by TheAirBlow** [comment]
I'm currently working on my own solution called `remEDIFIER`. 

After doing some reverse-engineering, I found out you can get the feature flags by sending `D8`. It's values are parsed by the Edifier ConneX app inside `com.edifier.lib_connect.models.D8Bean`.

**Comment by wh201906** [comment]
Never tried Edifier ConneX app before. Thanks for your information! @TheAirBlow

**Comment by TheAirBlow** [comment]
> Never tried Edifier ConneX app before. Thanks for your information! @TheAirBlow

Is Edifier Connect obfuscated? I would assume both of them would use the same libraries, as `lib_connect`, `lib_common` and others would suggest.

**Comment by TheAirBlow** [comment]
Also, the way the app identifies the headphones without communicating directly is getting the first service UUID from `EDIFIER BLE` device, over which you can also communicate when the device is connected over USB. Both the Bluetooth device and the BLE device also have the same mac address.

**Comment by wh201906** [comment]
> Is Edifier Connect obfuscated?

I'm not sure. I get the command set by checking the Logcat output and Edifier Connect prints every command there. Does Edifier ConneX print the command in the Logcat output?

**Comment by BreakpointDebug1234** [comment]
> Thanks for your work! I can add a dedicated profile for W820NB before merging this PR. According to some reviews, it seems W820NB doesn't support Sound Effect so it's reasonable these functions don't work.
> 
> I also found the `Control Settings` panel might be unavailable for W820NB. Could you please help me to test it? (For W820NB Double Gold, when clicking the Bluetooth button, the modes will only switch among the selected modes in the `Control Settings`)

Tested and looks like `Control Settings` doesn't do anything as well but that can be disabled too.

**Comment by wh201906** [comment]
> Tested and looks like `Control Settings` doesn't do anything as well but that can be disabled too.

So no matter which modes are selected in `Control Settings`, the modes will only switch to a fixed set of them when pressing the Bluetooth button, right?
Could you please tell me what will hear when pressing the Bluetooth button? Does the sequence change if you select different modes in `Control Settings`?

**Comment by BreakpointDebug1234** [comment]
> > Tested and looks like `Control Settings` doesn't do anything as well but that can be disabled too.
> 
> So no matter which modes are selected in `Control Settings`, the modes will only switch to a fixed set of them when pressing the Bluetooth button, right? Could you please tell me what will you hear when pressing the Bluetooth button? Does the sequence change if you select different modes in `Control Settings?`

Oh, I now understand what's supposed to do. Sadly I can't use the physical buttons on the headphones due a hardware issue (volume down button has broken and for some reason that renders the Bluetooth button unusable too) 

Btw, thanks for the app, I can now get around these issues without having to use the edifer app plus I can use it on PC too.

**Comment by wh201906** [comment]
I see. I will also disable the `Control Settings` for W820NB. If someone reports it is available, I will enable it again.

**Comment by TheAirBlow** [comment]
> > Is Edifier Connect obfuscated?
> 
> I'm not sure. I get the command set by checking the Logcat output and Edifier Connect prints every command there. Does Edifier ConneX print the command in the Logcat output?

Didn't check. I did decompile it with Jadx though, and looks like there is a lot of debug logging in here.

**Comment by TheAirBlow** [comment]
It's probably better to just decompile it and look through the source code than trying to crack your head trying to understand what all of those binary packets mean.

File: Qt/devices/deviceinfo.json [hunk 1]
```diff
@@ -1,9 +1,18 @@
 {
-    "basedevice":{
+    "basedevice": {
         "Name": "Generic Device",
         "MaxNameLength": 24,
         "HiddenFeatures": []
     },
+    "w820": {
```
Comment (wh201906) [inline Qt/devices/deviceinfo.json]: Thanks! Please use a similar name style here.
```suggestion
    "w820nb": {
```

File: README_zh_CN.md [hunk 2]
```diff
@@ -15,8 +15,8 @@
 + 从文件导入设置
 
 # 缺点
-+ 仅测试过 W820NB双金标版 和 W200BT Plus  
-  (也许能用在 W820NB Plus 和 W820NB 上)  
++ 仅测试了 W820NB、W820NB双金标版 和 W200BT Plus
```
Comment (wh201906) [inline README_zh_CN.md]: Thank you for using google translate for it! However, I prefer to not make unnecessary changes here.
```suggestion
+ 仅测试过 W820NB、W820NB双金标版 和 W200BT Plus  
```

File: README_zh_CN.md [hunk 3]
```diff
@@ -15,8 +15,8 @@
...
+  (也许可以在 W820NB Plus 上使用)
```
Comment (wh201906) [inline README_zh_CN.md]: For the same reason above
```suggestion
  (也许能用在 W820NB Plus 上)  
```

**Review by wh201906 — CHANGES_REQUESTED** [review]
Thanks for your work!

**Review by wh201906 — APPROVED** [review]
Great. Thanks for your work!

**Pull request merged**